internal/plugins/darwin/hostinfo.go

```diff
@@ -247,72 +254,50 @@ func (self *HostinfoPlugin) setCloudMetadata(data *HostInfoData) (err error) {
 
 // hardwareOverview structure keeps the values extracted from osX system_profiler.
 type hardwareOverview struct {
-	ModelName          string
-	ModelIdentifier    string
+	ModelName       string
+	ModelIdentifier string
+	Memory          string
+	HardwareUUID    string
+	processorInfo
+}
+
+// processorInfo structure keeps the values extracted from osX system_profiler
+// for the processor information.
+type processorInfo struct {
 	ProcessorName      string
 	ProcessorSpeed     string
-	Memory             string
 	NumberOfProcessors string
 	TotalNumberOfCores string
-	HardwareUUID       string
 }
 
 func (hip *HostinfoPlugin) getHardwareOverview() (hardwareOverview, error) {
-	result := hardwareOverview{}
-
 	out, err := hip.readDataFromCmd("system_profiler", "SPHardwareDataType")
 	if err != nil {
-		return result, err
+		return hardwareOverview{}, err
 	}
 
-	for _, line := range strings.Split(out, "\n") {
-		line = strings.TrimSpace(line)
-
-		if strings.HasPrefix(line, "Model Name: ") {
-			result.ModelName = strings.TrimPrefix(line, "Model Name: ")
-		}
-
-		if strings.HasPrefix(line, "Model Identifier: ") {
-			result.ModelIdentifier = strings.TrimPrefix(line, "Model Identifier: ")
-		}
-
-		if strings.HasPrefix(line, "Processor Name: ") {
-			result.ProcessorName = strings.TrimPrefix(line, "Processor Name: ")
-		}
-
-		if strings.HasPrefix(line, "Processor Speed: ") {
-			result.ProcessorSpeed = strings.TrimPrefix(line, "Processor Speed: ")
-		}
+	cpuInfo := getProcessorData(out)
 
-		if strings.HasPrefix(line, "Number of Processors: ") {
-			result.NumberOfProcessors = strings.TrimPrefix(line, "Number of Processors: ")
-		}
-
-		if strings.HasPrefix(line, "Total Number of Cores: ") {
-			result.TotalNumberOfCores = strings.TrimPrefix(line, "Total Number of Cores: ")
-		}
-
-		if strings.HasPrefix(line, "Hardware UUID: ") {
-			uuid := strings.TrimPrefix(line, "Hardware UUID: ")
-			if !validateHardwareUUID(uuid) {
-				hlog.WithError(err).Debug("Error detecting hardware UUID")
-				result.HardwareUUID = "unknown"
-			} else {
-				result.HardwareUUID = uuid
-			}
-		}
+	memory, err := memoryToKb(helpers.SplitRightSubstring(out, "Memory: ", "\n"))
+	if err != nil {
+		hlog.WithFields(logrus.Fields{
+			"line": helpers.SplitRightSubstring(out, "Memory: ", "\n"),
+		}).Debug("Unexpected format for 'Memory' field.")
+	}
 
-		if strings.HasPrefix(line, "Memory: ") {
-			result.Memory, err = memoryToKb(strings.TrimPrefix(line, "Memory: "))
-			if err != nil {
-				hlog.WithFields(logrus.Fields{
-					"line": line,
-				}).Debug("Unexpected format for 'Memory' field.")
-			}
-		}
+	uuid := helpers.SplitRightSubstring(out, "Hardware UUID: ", "\n")
+	if !validateHardwareUUID(uuid) {
+		hlog.WithError(err).Debug("Error detecting hardware UUID")
+		uuid = "unknown"
 	}
 
-	return result, nil
+	return hardwareOverview{
```

Comment:
I would use named fields by I guess is matter of preference. Adding it for the tech ☕

Reply:
Good point for the tech ☕ ! In this case I leave it without fields names because as being a composed struct, I would have need to set the struct fields in two different code sections.